### PR TITLE
amazon/ec2_vpc_route_table: Fix bug in counting subnets by 'Name' tag

### LINKED ISSUE #1551: ec2_vpc_route_table "Multiple subnets named XXX"
Hi,
thanks for your ec2_vpc_route_table module but i noticed a bug anoying with subnet attribute.

I use the following task:

```
- name: Create routing table
  ec2_vpc_route_table:
    vpc_id: "{{ item.0.vpc_id }}"
    region: "{{ item.0.region }}"
    tags: "{{ item.1.tags }}"
    routes: "{{ item.1.routes }}"
    subnets: "{{ item.1.subnets }}"
  with_subelements:
    - amazon.routes
    - config
```

Here is the vars file:

```
amazon:
  routes:
    vpc_test:
      vpc_id: "vpc-aaaaaaaa"
      region: "eu-wtf-1"
      config:
        - tags:
            Name: "test-public"
          subnets:
            - "subnet-XXX"
            - "subnet-YYY"
            - "subnet-ZZZ"
          routes:
            - { dest: "172.16.0.0/16", gateway_id: "vgw-XXX" }
            - { dest: "192.168.1.0/24", gateway_id: "vgw-XXX" }
```

This works with the following syntax with subnet IDs but if i replace subnet-XXX by names, it doesn't work with multiple subnets.

```
failed: [127.0.0.1] => (item=({u'vpc_id': u'vpc-XXX', u'region': u'eu-central-1'}, {u'routes': [{u'dest': u'172.16.0.0/16', u'gateway_id': u'vgw-XXX'}, {u'dest': u'192.168.1.0/24', u'gateway_id': u'vgw-XXX'}], u'subnets': [u'protect-test-a', u'protect-test2-a', u'protect-test3-a'], u'tags': {u'Name': u'test-public'}})) => {"failed": true, "item": [{"region": "eu-wtf-1", "vpc_id": "vpc-XXX"}, {"routes": [{"dest": "172.20.0.0/16", "gateway_id": "vgw-XXX"}, {"dest": "192.168.1.0/24", "gateway_id": "vgw-XXX"}], "subnets": ["protect-test-a", "protect-test2-a", "protect-test3-a"], "tags": {"Name": "test-public"}}], "msg": "Multiple subnets named \"protect-test-a\""}
```

Thanks for your reading

### PR DESCRIPTION
**Pull request by gechr**
Fixes #1551 

---

Current logic:

``` python
len([True, True, False, False, False])  # 5
```

New logic:

``` python
len([1, 1])  # 2
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @gechr. @erydo @wimnat please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by robynbergeron** [comment]
@erydo @wimnat This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
@erydo @wimnat still waiting on your review.  Please comment with text 'shipit' or 'needs_revision' as appropriate. If we don't hear from you within 14 days, we will start to look for additional maintainers for this module.

**Comment by Constantin07** [comment]
@gechr Many Thanks. 
Tested this fix and it works just fine.
Any ETA on when it's going to be reviewed and merged?

**Comment by wimnat** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @gechr for this PR, and thanks @erydo, @wimnat for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @gechr for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by zx8** [comment]
ready_for_review

**Comment by gechr** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @gechr. To the current maintainers, @erydo, @wimnat please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Kicking this stalled PR.

**Comment by gregdek** [comment]
Thanks @gechr. To the current maintainers, @erydo, @wimnat please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by wimnat** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @gechr for this PR, and thanks @erydo, @wimnat for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/ec2_vpc_route_table.py [hunk 1]
```diff
@@ -183,11 +183,11 @@ def find_subnets(vpc_conn, vpc_id, identified_subnets):
             filters={'vpc_id': vpc_id, 'tag:Name': subnet_names})
 
         for name in subnet_names:
-            matching = [s.tags.get('Name') == name for s in subnets_by_name]
-            if len(matching) == 0:
+            matching_count = len([1 for s in subnets_by_name if s.tags.get('Name') == name])
```
Comment (rwe) [inline cloud/amazon/ec2_vpc_route_table.py:186]: Sorry for missing this review! This change looks good; the original typo looks like it actually just meant to use a condition on the generator:

``` py
matching = [s for s in subnets_by_name if s.tags.get('Name') == name]
```

**Comment by alikins** [comment]
Is the comment in https://github.com/ansible/ansible-modules-extras/pull/1643#discussion_r69940753 sufficient to reconsider this change?

**Comment by rwe** [comment]
@alikins - My comment shouldn't be considered a blocker.

**Comment by gregdek** [comment]
@gechr A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@gechr Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gechr** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @gechr. To the current maintainers, @erydo, @wimnat please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by wimnat** [comment]
For the third time, shipit

**Comment by gregdek** [comment]
Thanks again to @gechr for this PR, and thanks @erydo, @wimnat for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**